ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -207,10 +207,37 @@ protected void InitializeHttpClient(HttpClientHandler httpClientHandler, params
             FirstMessageHandler = currentHandler;
             HttpClient = newClient;
             Type type = this.GetType();
-            HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(type.FullName,
-                GetClientVersion()));
+            //setting userAgentBelow is removed because now the client can set it using SetUserAgent method
```

Comment:
This would be a breaking change for clients that are not, or don't need to, set their own UserAgent values.  It would be better to continue setting the default values as it does today and provide a way of overriding it for clients that want to.